nimbus-primitives/src/lib.rs

```diff
@@ -35,6 +36,19 @@ pub use digests::CompatibleDigestItem;
 
 pub use inherents::{InherentDataProvider, INHERENT_IDENTIFIER};
 
+/// Input NimbusId and returns the additional inherent digests
+pub trait InherentDigestsProvider<Id> {
+	type Digests: IntoIterator<Item = DigestItem>;
+	fn provide_inherent_digests(&self, id: Id) -> Self::Digests;
+}
+
+impl<Id> InherentDigestsProvider<Id> for () {
+	type Digests = Vec<DigestItem>;
+	fn provide_inherent_digests(&self, _id: Id) -> Self::Digests {
+		Vec::new()
+	}
+}
```

Comment:
Thanks @librelois !

Not sure how to implement this without allocating some vec

An alternative could be
```suggestion
impl<Id> InherentDigestsProvider<Id> for () {
	type Digests = DigestItem;
	fn provide_inherent_digests(&self, _id: Id) -> Self::Digests {
		DigestItem::Other(Vec::new())
	}
}
```
but I don't think this is better

Reply:
Maybe with the type `[DigestItem: 0]` or `&'static [DigestItem]`